cmd/cue/cmd/get_go.go

```diff
@@ -494,25 +528,24 @@ func (e *extractor) extractPkg(root string, p *packages.Package) error {
 			e.pkgNames[pkgPath] = info
 		}
 
-		decls := []cueast.Decl{}
+		var fileDecls []cueast.Decl
 		for _, d := range f.Decls {
 			switch d := d.(type) {
 			case *ast.GenDecl:
-				decls = append(decls, e.reportDecl(d)...)
+				fileDecls = append(fileDecls, e.reportDecl(d)...)
 			}
 		}
 
-		if len(decls) == 0 && f.Doc == nil {
+		if len(fileDecls) == 0 && f.Doc == nil && !flagOutFile.IsSet(e.cmd) {
 			continue
 		}
+		decls = append(decls, fileDecls...)
 
-		pName := flagPackage.String(e.cmd)
-		if pName == "" {
-			pName = p.Name
-		}
+		addDoc(f.Doc, cuePkg)
 
-		pkg := &cueast.Package{Name: e.ident(pName, false)}
-		addDoc(f.Doc, pkg)
+		if flagOutFile.IsSet(e.cmd) && i != len(p.Syntax)-1 {
```

Comment:
pretty funky. the logic is not particularly easy to follow. can we add more comments? perhaps one jsut above the main for loop, explaining how the regular and "outfile" modes differ.

Reply:
Yeah. The logic is kinda ugly but I stared the last two hours on this and didn't found a nice way to structure this that wouldn't copy all the logic